src/jit/importer.cpp

```diff
@@ -14078,7 +14208,20 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                 op3 = impPopStack().val; // Size
                 op2 = impPopStack().val; // Src
                 op1 = impPopStack().val; // Dest
-                op1 = gtNewBlkOpNode(GT_COPYBLK, op1, op2, op3, (prefixFlags & PREFIX_VOLATILE) != 0);
+
+                if (op3->IsCnsIntOrI())
+                {
+                    size = (unsigned)op3->AsIntConCommon()->IconValue();
+                    op1  = new (this, GT_BLK) GenTreeBlk(GT_BLK, TYP_STRUCT, op1, size);
+                }
+                else
+                {
+                    op1  = new (this, GT_DYN_BLK) GenTreeDynBlk(op1, op3);
+                    size = 0;
+                }
+                op2 = gtNewOperNode(GT_IND, TYP_STRUCT, op2);
```

Comment:
Yes, I think it's safe to eliminate this one.

---
In reply to: [77397229](https://github.com/dotnet/coreclr/pull/7027#discussion_r77397229) [](ancestors = 77397229)